src/clib/lib/enkf/model_config.cpp

```diff
@@ -426,7 +420,11 @@ void model_config_init(model_config_type *model_config,
     if (config_content_has_item(config, HISTORY_SOURCE_KEY)) {
         const char *history_source =
             config_content_iget(config, HISTORY_SOURCE_KEY, 0, 0);
-        source_type = history_get_source_type(history_source);
+        history_source_type source_type = REFCASE_HISTORY;
```

Comment:
Ah, I found it. You are shadowing source_type here. That is why clang-tidy is complaining

Reply:
I think you forgot to fix this.

Reply:
You are right! Rebase gone wrong, I removed it in the first commit but reintroduced it later :grimacing: fixed now

Reply:
Not sure it would catch this, as it did not change the defaults, but I have now added an integration test for the three options.

Reply:
The thing is that your code shadowed the variable so that history_source was never REFCASE_SIMULATED, but the tests did not fail.